src/ipc_bridge.rs

```diff
@@ -15,46 +19,122 @@ pub struct ClientId {
     pub token: mio::Token,
 }
 
-struct Connection {
+// We abstract over unix and TCP connections. Since receiver and sender both get a copy of the
+// socket, we need to clone them. Since we store them in slab (which means the trait cannot be
+// sized), we have to return boxes too.
+trait MioStream: Send + io::Read + io::Write + mio::Evented {
+    fn try_clone(&self) -> io::Result<Box<MioStream>>;
+}
```

Comment:
Yes, it does. I was unable to combine the two of them though. This one is not Sized and the try_clone() methods return `Box<MioStream>` instead of Self. Any idea how to consolidate the two into one is greatly appreciated - somehow I did not wrap my head around the trait system completely yet it seems.